test/SILGen/typed_throws.swift

```diff
@@ -315,3 +315,11 @@ class MySubclass: MyClass {
 class MySubsubclass: MySubclass {
   override func f() { }
 }
+
+// Check that SILGen does not crash because of a parenthesized try expression
+class Class {
```

Comment:
Since this is checking a crasher I'd suggest this to be moved to `validation-test/compiler_crashers_fixed/` or `validation-test/SILGen/` as `issue-73244.swift` with `// RUN: %target-swift-emit-silgen %s -verify`